website2/website/versioned_docs/version-0.20.2-incubating/compiling-code-organization.md

```diff
@@ -0,0 +1,199 @@
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+-->
+
+This document contains information about the Heron codebase intended primarily
+for developers who want to contribute to Heron. The Heron codebase lives on
+[github]({{% githubMaster %}}).
+
+If you're looking for documentation about developing topologies for a Heron
+cluster, see [Building Topologies](topology-development-topology-api-java) instead.
+
+## Languages
+
+The primary programming languages for Heron are C++, Java, and Python.
+
+* **C++ 11** is used for most of Heron's core components, including the
+[Topology Master](heron-architecture#topology-master), and
+[Stream Manager](heron-architecture#stream-manager).
+
+* **Java 11** is used primarily for Heron's [topology
+API](heron-topology-concepts), and [Heron Instance](heron-architecture#heron-instance).
+It is currently the only language in which topologies can be written. Instructions can be found
+in [Building Topologies](../../developers/java/topologies), while documentation for the Java
+API can be found [here](/api/org/apache/heron/api/topology/package-summary.html). Please note that Heron topologies do not require Java 11 and can be written in Java 7 or later.
+
+* **Python 2** (specifically 2.7) is used primarily for Heron's [CLI interface](user-manuals-heron-cli) and UI components such as [Heron UI](user-manuals-heron-ui) and the [Heron Tracker](user-manuals-heron-tracker-runbook).
+
+## Main Tools
+
+* **Build tool** --- Heron uses [Bazel](http://bazel.io/) as its build tool.
+Information on setting up and using Bazel for Heron can be found in [Compiling Heron](compiling-overview).
+
+* **Inter-component communication** --- Heron uses [Protocol
+Buffers](https://developers.google.com/protocol-buffers/?hl=en) for
+communication between components. Most `.proto` definition files can be found in
+[`heron/proto`]({{% githubMaster %}}/heron/proto).
+
+* **Cluster coordination** --- Heron relies heavily on ZooKeeper for cluster
+coordination for distributed deployment, be it for [Aurora](schedulers-aurora-cluster) or for a [custom
+scheduler](extending-heron-scheduler) that you build. More information on ZooKeeper
+components in the codebase can be found in the [State
+Management](#state-management) section below.
+
+## Common Utilities
+
+The [`heron/common`]({{% githubMaster %}}/heron/common) contains a variety of
```

Comment:
I suspect this links like this are broken, as the version interpolation is `{{heron:version}}` instead of `{{% heronpyVersion %}}`

Reply:
Thanks for catching.  Yes. this is left over from the old site.